### PR TITLE
next-redux-wrapper 업데이트

### PR DESCRIPTION
**Pull request by tirr-c**
- `next-redux-wrapper@^6.0.2`
- `next-redux-saga` 제거하고 saga 관련 코드 추가 (짧습니다)

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -48,6 +44,10 @@ class StoreApp extends App<StoreAppProps> {
     const pageProps = Component.getInitialProps
       ? await Component.getInitialProps(ctx)
       : {};
+    if (ctx.req) {
+      ctx.store.dispatch(END);
+      await (ctx.store as any).sagaTask.toPromise();
+    }
```
Comment (jeong-sik) [inline src/pages/_app.tsx:50]: 궁금한 부분이라 찾아보다 링크 해 놓았습니다!
https://github.com/kirill-konshin/next-redux-wrapper#using-pages_app

File: src/services/books/reducer.ts [hunk 2]
```diff
@@ -82,5 +84,17 @@ export class BooksReducer extends ImmerReducer<BooksState> {
   }
 }
 
-export const booksReducer = createReducerFunction(BooksReducer, booksInitialState);
+const innerBooksReducer = createReducerFunction(BooksReducer, booksInitialState);
+export function booksReducer(
+  state: BooksState = booksInitialState,
+  action: AnyAction,
+): BooksState {
+  if (action.type === HYDRATE) {
```
Comment (jeong-sik) [inline src/services/books/reducer.ts:92]: `Please note that your reducer must have the HYDRATE action handler. HYDRATE action handler must properly reconciliate the hydrated state on top of the existing state (if any). This behavior was added in version 6 of this library.`
https://github.com/kirill-konshin/next-redux-wrapper#usage
`HYDRATE`

File: src/pages/[genre]/index.tsx [hunk 3]
```diff
@@ -144,8 +144,8 @@ Home.getInitialProps = async (ctx: ConnectedInitializeProps) => {
     query,
     res,
     store,
-    isServer,
   } = ctx;
+  const isServer = Boolean(ctx.req);
```
Comment (J132134) [inline src/pages/[genre]/index.tsx:148]: `ctx.isServer` 는 왜 이렇게 변경되었나요?

Comment (J132134) [inline src/pages/[genre]/index.tsx:148]: ctx.isServer 가 사라졌나요?

Comment (jeong-sik) [reply src/pages/[genre]/index.tsx:148]: 놀랍게도 원래 없었던 거 같기도 합니다.

https://github.com/vercel/next.js/issues/2946#issuecomment-329235139
https://github.com/vercel/next.js/blob/e125d905a0dd93d247c6122d349c2c90268f0713/packages/next/next-server/lib/utils.ts#L94

Comment (tirr-c) [reply src/pages/[genre]/index.tsx:148]: `next-redux-wrapper@5`까지 `isServer`가 있었다가 6에서 사라졌습니다.

**Pull request merged**